actors/common/src/main/java/org/sunbird/common/ShadowUserProcessor.java

```diff
@@ -0,0 +1,361 @@
+            filters.put(JsonKey.EXTERNAL_ID, shadowUser.getOrgExtId());
+            filters.put(JsonKey.CHANNEL, shadowUser.getChannel());
+            request.put(JsonKey.FILTERS, filters);
+            SearchDTO searchDTO = ElasticSearchHelper.createSearchDTO(request);
+            Map<String, Object> response = (Map<String, Object>) ElasticSearchHelper.getResponseFromFuture(elasticSearchService.search(searchDTO, ProjectUtil.EsType.organisation.getTypeName()));
+            List<Map<String, Object>> orgData = ((List<Map<String, Object>>) response.get(JsonKey.CONTENT));
+            if (CollectionUtils.isNotEmpty(orgData)) {
+                Map<String, Object> orgMap = orgData.get(0);
+                return (String) orgMap.get(JsonKey.ID);
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+
+    private List<String> getOrganisationIds(Map<String, Object> dbUser) {
+        List<String> organisationsIds = new ArrayList<>();
+        ((List<Map<String, Object>>) dbUser.get(JsonKey.ORGANISATIONS)).stream().forEach(organisation -> {
+            organisationsIds.add((String) organisation.get(JsonKey.ORGANISATION_ID));
+        });
+        return organisationsIds;
+    }
+
+
+
+
+    private void syncUserToES(String userId) {
+        Map<String, Object> fullUserDetails = Util.getUserDetails(userId, null);
+        try {
+            Future<Boolean> future = elasticSearchService.update(JsonKey.USER, userId, fullUserDetails);
+            if ((boolean) ElasticSearchHelper.getResponseFromFuture(future)) {
+                ProjectLogger.log("ShadowUserMigrationScheduler:updateUserStatus: data successfully updated to elastic search with userId:".concat(userId + ""), LoggerEnum.INFO.name());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            ProjectLogger.log("ShadowUserMigrationScheduler:syncUserToES: data failed to updates in elastic search with userId:".concat(userId + ""), LoggerEnum.ERROR.name());
+        }
+    }
+
+
+    private void deleteUserOrganisations(Map<String, Object> esUserMap) {
+        ((List<Map<String, Object>>) esUserMap.get(JsonKey.ORGANISATIONS)).stream().forEach(organisation -> {
+            String id = (String) organisation.get(JsonKey.ID);
+            deleteOrgFromUserOrg(id);
+        });
+    }
+
+    private void deleteOrgFromUserOrg(String id) {
+        Response response = cassandraOperation.deleteRecord(JsonKey.SUNBIRD, JsonKey.USER_ORG, id);
+        ProjectLogger.log("ShadowUserProcessor:deleteOrgFromUserOrg:user org is deleted ".concat(response.getResult() + ""));
+    }
+
+
+    private void registerUserToOrg(String userId, String organisationId) {
+        Map<String, Object> reqMap = new WeakHashMap<>();
+        List<String> roles = new ArrayList<>();
+        roles.add(ProjectUtil.UserRole.PUBLIC.getValue());
+        reqMap.put(JsonKey.ROLES, roles);
+        String hashTagId = Util.getHashTagIdFromOrgId(organisationId);
```

Comment:
We can use loacl cache for orgId and hashtahId

Reply:
done